test/functional/user_sessions_controller_test.rb

```diff
@@ -267,6 +267,18 @@ class UserSessionsControllerTest < ActionController::TestCase
     assert_equal "Successfully logged out.",  flash[:notice]
   end
   
+  test "logging in with banned user through oauth should fail and redirect correctly" do
```

Comment:
Write one test for moderator and one test for banned user. So, one test is missing.